pkg/hintrunner/zero/zerohint_dictionaries.go

```diff
@@ -10,6 +10,122 @@ import (
+func newDefaultDictNewHint(defaultValue hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DefaultDictNew",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> if '__dict_manager' not in globals():
+			//> 	from starkware.cairo.common.dict import DictManager
+			//> 	__dict_manager = DictManager()
+			//>
+			//> memory[ap] = __dict_manager.new_default_dict(segments, ids.default_value)
+
+			//> if '__dict_manager' not in globals():
+			//> 	from starkware.cairo.common.dict import DictManager
+			//> 	__dict_manager = DictManager()
+			dictionaryManager, ok := ctx.ScopeManager.GetZeroDictionaryManager()
+			if !ok {
+				dictionaryManager = hinter.NewZeroDictionaryManager()
+				err := ctx.ScopeManager.AssignVariable("__dict_manager", dictionaryManager)
+				if err != nil {
+					return err
+				}
+			}
+
+			//> memory[ap] = __dict_manager.new_default_dict(segments, ids.default_value)
+			defaultValue, err := hinter.ResolveAsFelt(vm, defaultValue)
+			if err != nil {
+				return err
+			}
+			defaultValueMv := memory.MemoryValueFromFieldElement(defaultValue)
+			newDefaultDictionaryAddr := dictionaryManager.NewDefaultDictionary(vm, defaultValueMv)
+			newDefaultDictionaryAddrMv := memory.MemoryValueFromMemoryAddress(&newDefaultDictionaryAddr)
+			apAddr := vm.Context.AddressAp()
+			return vm.Memory.WriteToAddress(&apAddr, &newDefaultDictionaryAddrMv)
+		},
+	}
+}
+
+func createDefaultDictNewHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+	defaultValue, err := resolver.GetResOperander("default_value")
+	if err != nil {
+		return nil, err
+	}
+	return newDefaultDictNewHint(defaultValue), nil
+}
+
+// DictRead hint reads the result of given key from given dictionary
+// and writes it to value
+//
+// `newDictReadHint` takes 3 operander as argument
+//   - `dictPtr` variable will be pointer to the dictionary to read from
+//   - `key` variable will be the key read from the dictionary
+//   - `value` variable will be the variable to which the result of
+//     the `key` read from the dictionary pointed to by `dictPtr`
+//     will be written to
+func newDictReadHint(dictPtr, key, value hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DictRead",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> dict_tracker = __dict_manager.get_tracker(ids.dict_ptr)
+			//> dict_tracker.current_ptr += ids.DictAccess.SIZE
+			//> ids.value = dict_tracker.data[ids.key]
+
+			//> dict_tracker = __dict_manager.get_tracker(ids.dict_ptr)
+			dictPtr, err := hinter.ResolveAsAddress(vm, dictPtr)
+			if err != nil {
+				return err
+			}
+			dictionaryManager, ok := ctx.ScopeManager.GetZeroDictionaryManager()
+			if !ok {
+				return fmt.Errorf("__dict_manager not in scope")
+			}
+
+			//> dict_tracker.current_ptr += ids.DictAccess.SIZE
+			err = dictionaryManager.IncrementFreeOffset(*dictPtr, 3)
```

Comment:
are you sure that incrementing the free offset is enough? dont we need to also write the entry in the dict memory segment?

Reply:
lgtm for the rest

Reply:
Yeah thats all the python code does!

Memory is updated as part of the main `dict_read` cairo function: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/dict.cairo#L22